### PR TITLE
doc updates for v2.4.0

### PR DESCRIPTION
**Pull request by kishorj**
### Checklist
- [ ] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: docs/guide/service/nlb.md [hunk 1]
```diff
@@ -59,6 +59,8 @@ NLB IP mode is determined based on the `service.beta.kubernetes.io/aws-load-bala
 
 ### Instance mode
 Similar to the IP mode, the instance mode is based on the annotation `service.beta.kubernetes.io/aws-load-balancer-nlb-target-type` value `instance`. Here is a sample manifest snippet:
+!!!warning "NodePort allocation"
+    k8s version 1.22 and later support disabling NodePort allocation by setting the service field `spec.allocateLoadBalancerNodePorts` to `true`. If the NodePort is not allocated for a service port, the controller will fail to reconcile instance mode NLB.
```
Comment (M00nF1sh) [inline docs/guide/service/nlb.md:63]: `true`-> false?

File: docs/guide/service/annotations.md [hunk 2]
```diff
@@ -83,9 +83,13 @@ Traffic Routing can be controlled with following annotations:
     - `instance` mode will route traffic to all EC2 instances within cluster on the [NodePort](https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport) opened for your service.
 
         !!!note ""
-            service must be of type `NodePort` or `LoadBalancer` for `instance` targets
+            - service must be of type `NodePort` or `LoadBalancer` for `instance` targets
+            - for k8s 1.22 and later if `spec.allocateLoadBalancerNodePorts` is set to `true`, `NodePort` must be allocated manually
```
Comment (M00nF1sh) [inline docs/guide/service/annotations.md:87]: `true`-> false?

File: docs/guide/service/nlb.md [hunk 3]
```diff
@@ -95,3 +97,10 @@ security groups, the controller expects only one security group tagged with the
 | `kubernetes.io/cluster/${cluster-name}` | `owned` or `shared`   |
 
 `${cluster-name}` is the name of the kubernetes cluster
+
+## Load Balancer Class
+The AWS Load Balancer Controller supports `LoadBalancerClass` starting v2.4.0 release on k8s 1.22 or later clusters. The LoadBalancerClass provides a cloudprovider agnostic way of offloading the load balancer reconciliation to an external controller. This controller uses the `service.k8s.aws/nlb` as the default class,
+you can configure it to a different value via the controller flag `--load-balancer-class`.
+
+When you specify the `spec.loadBalancerClass` on a service of type `LoadBalancer` during service creation, this controller creates an internal NLB with instance targets by default. If the LoadBalancerClass is not the configured for this controller, this controller ignores the servive resource completely regardless of the annotation
```
Comment (M00nF1sh) [inline docs/guide/service/nlb.md:105]: s/servive/service

File: docs/guide/service/nlb.md (hunk 1 above)
Comment (kishorj) [reply docs/guide/service/nlb.md]: thanks

File: docs/guide/service/annotations.md (hunk 2 above)
Comment (kishorj) [reply docs/guide/service/annotations.md]: updated

**Review by M00nF1sh — APPROVED** [review]
/lgtm

**Review by orsenthil — COMMENTED** [review]
LGTM.

**Pull request merged**